File: apps/web/src/hooks/api/career.ts
import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query';
import apiClient from '@/lib/api-client';

// TypeScript interfaces
export interface HealthBreakdownItem {
  score: number;
  weight: number;
  reasoning: string;
}

export interface HealthScore {
  overall_score: number;
  breakdown: Record<string, HealthBreakdownItem>;
  grade: string;
  trend: string;
}

export interface Insight {
  id: string;
  type: string;
  title: string;
  description: string;
  category: string;
  severity: string;
  confidence: number;
  evidence: string[];
  actionable: boolean;
  created_at: string;
}

export interface Recommendation {
  id: string;
  title: string;
  description: string;
  category: string;
  priority: string;
  difficulty: string;
  estimated_time: string;
  expected_benefit: string;
  confidence: number;
  evidence: string[];
  status: string;
  user_status: string;
  completed_at: string | null;
  created_at: string;
}

export interface Prediction {
  id: string;
  type: string;
  title: string;
  description: string;
  predicted_value: number;
  confidence: number;
  time_horizon: string;
  factors: string[];
  created_at: string;
}

export interface Opportunity {
  type: string;
  title: string;
  description: string;
  confidence: number;
  evidence: string[];
}

export interface CareerGoal {
  id: string;
  title: string;
  description: string | null;
  category: string;
  target_date: string | null;
  status: string;
  priority: string;
  progress: number;
  created_at: string;
}

export interface CareerProfile {
  id: string;
  user_id: string;
  current_role: string | null;
  target_role: string | null;
  target_industry: string | null;
  target_seniority: string | null;
  salary_expectation_min: number | null;
  salary_expectation_max: number | null;
  preferred_locations: string[] | null;
  remote_preference: string | null;
  work_style: string | null;
  career_stage: string | null;
  years_experience: number | null;
  primary_domain: string | null;
  secondary_domains: string[] | null;
  career_focus_areas: string[] | null;
  learning_style: string | null;
  risk_tolerance: string | null;
  growth_priority: string | null;
  work_life_balance_priority: string | null;
  company_size_preference: string | null;
  company_type_preference: string | null;
  created_at: string;
  updated_at: string;
}

export interface TodayPriority {
  title: string;
  description: string;
  estimated_time: string;
  expected_benefit: string;
  confidence: number;
}

export interface DashboardData {
  greeting: string;
  health_score: HealthScore;
  todays_priorities: TodayPriority[];
  insights: Insight[];
  recommendations: Recommendation[];
  predictions: Prediction[];
  opportunities: Opportunity[];
  goals: CareerGoal[];
  profile: CareerProfile;
}

// API hooks
export function useCareerDashboard() {
  return useQuery({
    queryKey: ['career', 'dashboard'],
    queryFn: () => apiClient.get<DashboardData>('/api/v1/career/dashboard'),
    staleTime: 5 * 60 * 1000,
  });
}

export function useCareerHealth() {
  return useQuery({
    queryKey: ['career', 'health'],
    queryFn: () => apiClient.get<HealthScore>('/api/v1/career/health'),
    staleTime: 5 * 60 * 1000,
  });
}

export function useUpdateCareerProfile() {
  const queryClient = useQueryClient();

  return useMutation({
    mutationFn: (profileData: Partial<CareerProfile>) =>
      apiClient.patch<CareerProfile>('/api/v1/career/profile', profileData),
    onSuccess: () => {
      queryClient.invalidateQueries({ queryKey: ['career'] });
    },
  });
}

export function useCreateCareerGoal() {
  const queryClient = useQueryClient();

  return useMutation({
    mutationFn: (goalData: {
      title: string;
      description?: string;
      category: string;
      target_date?: string;
      priority: string;
    }) => apiClient.post<{ id: string; message: string }>('/api/v1/career/goals', goalData),
    onSuccess: () => {
      queryClient.invalidateQueries({ queryKey: ['career'] });
    },
  });
}

export function useUpdateRecommendation() {
  const queryClient = useQueryClient();

  return useMutation({
    mutationFn: ({
      recommendationId,
      feedback,
    }: {
      recommendationId: string;
      feedback: { status: string; user_status: string; feedback?: string };
    }) =>
      apiClient.patch(`/api/v1/career/recommendations/${recommendationId}`, feedback),
    onSuccess: () => {
      queryClient.invalidateQueries({ queryKey: ['career'] });
    },
  });
}

File: apps/api/app/models/career.py
"""Pydantic models for career intelligence operations."""
from datetime import datetime
from typing import Any, Optional

from pydantic import BaseModel, Field


# Request Models
class CareerProfileUpdateRequest(BaseModel):
    """Request to update career profile."""

    current_role: Optional[str] = None
    target_role: Optional[str] = None
    target_industry: Optional[str] = None
    target_seniority: Optional[str] = None
    salary_expectation_min: Optional[int] = None
    salary_expectation_max: Optional[int] = None
    preferred_locations: Optional[list[str]] = None
    remote_preference: Optional[str] = None
    work_style: Optional[str] = None
    career_stage: Optional[str] = None
    years_experience: Optional[int] = None
    primary_domain: Optional[str] = None
    secondary_domains: Optional[list[str]] = None
    career_focus_areas: Optional[list[str]] = None
    learning_style: Optional[str] = None
    risk_tolerance: Optional[str] = None
    growth_priority: Optional[str] = None
    work_life_balance_priority: Optional[str] = None
    company_size_preference: Optional[str] = None
    company_type_preference: Optional[str] = None
    additional_preferences: Optional[dict[str, Any]] = None


class CareerGoalRequest(BaseModel):
    """Request to create career goal."""

    title: str = Field(..., min_length=1, max_length=255)
    description: Optional[str] = None
    category: str = Field(..., min_length=1, max_length=100)
    target_date: Optional[str] = None
    priority: str = Field(..., min_length=1, max_length=50)


class RecommendationFeedbackRequest(BaseModel):
    """Request to provide feedback on recommendation."""

    status: str = Field(..., min_length=1, max_length=50)
    user_status: str = Field(..., min_length=1, max_length=50)
    feedback: Optional[str] = None


# Response Models
class HealthBreakdownItem(BaseModel):
    """Health score breakdown item."""

    score: int
    weight: int
    reasoning: str


class HealthScoreResponse(BaseModel):
    """Career health score response."""

    overall_score: int
    breakdown: dict[str, HealthBreakdownItem]
    grade: str
    trend: str


class InsightResponse(BaseModel):
    """Career insight response."""

    id: str
    type: str
    title: str
    description: str
    category: str
    severity: str
    confidence: float
    evidence: list[str]
    actionable: bool
    created_at: datetime


class RecommendationResponse(BaseModel):
    """Career recommendation response."""

    id: str
    title: str
    description: str
    category: str
    priority: str
    difficulty: str
    estimated_time: str
    expected_benefit: str
    confidence: float
    evidence: list[str]
    status: str
    user_status: str
    completed_at: Optional[datetime]
    created_at: datetime


class PredictionResponse(BaseModel):
    """Career prediction response."""

    id: str
    type: str
    title: str
    description: str
    predicted_value: float
    confidence: float
    time_horizon: str
    factors: list[str]
    created_at: datetime


class OpportunityResponse(BaseModel):
    """Career opportunity response."""

    type: str
    title: str
    description: str
    confidence: float
    evidence: list[str]


class CareerGoalResponse(BaseModel):
    """Career goal response."""

    id: str
    title: str
    description: Optional[str]
    category: str
    target_date: Optional[str]
    status: str
    priority: str
    progress: int
    created_at: datetime


class CareerProfileResponse(BaseModel):
    """Career profile response."""

    id: str
    user_id: str
    current_role: Optional[str]
    target_role: Optional[str]
    target_industry: Optional[str]
    target_seniority: Optional[str]
    salary_expectation_min: Optional[int]
    salary_expectation_max: Optional[int]
    preferred_locations: Optional[list[str]]
    remote_preference: Optional[str]
    work_style: Optional[str]
    career_stage: Optional[str]
    years_experience: Optional[int]
    primary_domain: Optional[str]
    secondary_domains: Optional[list[str]]
    career_focus_areas: Optional[list[str]]
    learning_style: Optional[str]
    risk_tolerance: Optional[str]
    growth_priority: Optional[str]
    work_life_balance_priority: Optional[str]
    company_size_preference: Optional[str]
    company_type_preference: Optional[str]
    created_at: datetime
    updated_at: datetime


class TodayPriorityItem(BaseModel):
    """Today's priority item."""

    title: str
    description: str
    estimated_time: str
    expected_benefit: str
    confidence: float


class DashboardResponse(BaseModel):
    """Career dashboard response."""

    greeting: str
    health_score: HealthScoreResponse
    todays_priorities: list[TodayPriorityItem]
    insights: list[InsightResponse]
    recommendations: list[RecommendationResponse]
    predictions: list[PredictionResponse]
    opportunities: list[OpportunityResponse]
    goals: list[CareerGoalResponse]
    profile: CareerProfileResponse


class RecommendationUpdateResponse(BaseModel):
    """Response for recommendation update."""

    id: str
    status: str
    user_status: str
    feedback: Optional[str]
    completed_at: Optional[datetime]
